### ISSUE TITLE
Unexpected warning from no-undef when assigning to a variable declared in global scope

### ISSUE THREAD

**Issue by mamacdon** [eslint/eslint#283]
In the following code:

``` js
var a;
function f() {
    a = 1;
}
```

The `no-undef` rule incorrectly flags the code with `'a' is not defined`.
